cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -17,53 +15,234 @@
 import org.cyclonedx.generators.json.BomJsonGenerator;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * Command line tool to construct a CycloneDX SBOM.
  */
 public final class TemurinGenSBOM {
+
+    private static String fileName;
+
+    private static String name;
+    private static String version;
+    private static String value;
+    private static String url;
+    private static String hashes;
+    private static Bom bom;
+    private static  String cmd;
+    private static String comment;
+    private static Metadata meta = new Metadata();
+    private static Property prop1 = new Property();
+    private static Property prop2 = new Property();
+    private static Component comp = new Component();
+    private List<Component> comp1 = new ArrayList<>();
+    private static List<Property> prop = new ArrayList<>();
+    private static List<Hash> hash = new ArrayList<>();
+    private static Hash hash1 = new Hash(Hash.Algorithm.SHA3_256, hashes);
+    private static  ExternalReference extRef = new ExternalReference();
+    private static FileWriter file;
+
     private TemurinGenSBOM() {
     }
 
-    /**
-     * Main entry.
-     * @param args Arguments for sbom operation.
-     */
     public static void main(final String[] args) {
-        System.out.print("TemurinGenSBOM:");
-        for (String arg : args) {
-            System.out.print(" " + arg);
+
+
+        for (int i = 0; i < args.length; i++) {
+
+            if (args[i].equals("--jsonFile")) {
+                fileName = args[++i];
+            } else if (args[i].equals("--version")) {
+                version = args[++i];
+            } else if (args[i].equals("--name")) {
+                name = args[++i];
+            } else if (args[i].equals("--value")) {
+                value = args[++i];
+            } else if (args[i].equals("--url")) {
+                url = args[++i];
+            } else if (args[i].equals("--hashes")) {
+                hashes = args[++i];
+            } else if (args[i].equals("--comment")) {
+                comment = args[++i];
+            } else if (args[i].equals("--createNewSBOM")) {
+                cmd = "createNewSBOM";
+            } else if (args[i].equals("--addMetadata")) {        //This is Metadata Component. We can set "name" for Metadata->Component.
+                cmd = "addMetadata";
+            } else if (args[i].equals("--addMetadataProp")) {   //This is MetaData Component --> Property -> name-value: os, arch, variant, scmRef, buildRef, full_version_output, makejdk_any_platform_args
+                cmd = "addMetadataProperty";
+            } else if (args[i].equals("--addComponent")) {      //This is Components->Property: will add name-value.
+                cmd = "addComponent";
+            } else if (args[i].equals("--addComponentProp")) {  //This is Components->Property: will add name-value.
```

Comment:
You need to identify which Component Prop is being added to? maybe have a position parameter to the --addComponentProp arg which is the Component "name" to add prop to, eg:
--addComponentProp "thiscomp"

Reply:
@andrew-m-leonard I have tried making multiple Components by constructing multiple objects, but they all seem to add in same-one component. So far there is only one Metadata and one Component, further, both have different attributes.

Reply:
@andrew-m-leonard yes, working on it.

Reply:
don't worry about Props in this PR